exporter/sumologicexporter/exporter.go

```
@@ -16,44 +16,273 @@ package sumologicexporter
+	body := strings.Builder{}
+	var errs []error
+
+	if se.config.LogFormat == TextFormat {
+		// Concatenate log lines using `\n`
+		for j := 0; j < len(buffer); j++ {
+			nextLine := buffer[j].Body().StringVal()
+
+			if body.Len() > 0 && body.Len()+len(nextLine) > se.config.MaxRequestBodySize {
+				err := se.send(LogsPipeline, body.String(), fields)
+				if err != nil {
+					errs = append(errs, err)
+				}
+				body.Reset()
+			}
+
+			if body.Len() > 0 {
+				// Do not add newline after last line
+				body.WriteString("\n")
+			}
+
+			body.WriteString(nextLine)
+		}
+		err := se.send(LogsPipeline, body.String(), fields)
+
+		if err != nil {
+			errs = append(errs, err)
+		}
+
+	} else if se.config.LogFormat == JSONFormat {
+		for j := 0; j < len(buffer); j++ {
+			data := se.filterMetadata(buffer[j].Attributes(), true)
+			data[logKey] = buffer[j].Body().StringVal()
+			nextLine, _ := json.Marshal(data)
+
+			if body.Len() > 0 && body.Len()+len(nextLine) > se.config.MaxRequestBodySize {
+				err := se.send(LogsPipeline, body.String(), fields)
+				if err != nil {
+					errs = append(errs, err)
+				}
+				body.Reset()
+			}
+
+			if body.Len() > 0 {
+				// Do not add newline after last line
+				body.WriteString("\n")
+			}
+
+			body.Write(nextLine)
+		}
+		err := se.send(LogsPipeline, body.String(), fields)
+
+		if err != nil {
+			errs = append(errs, err)
+		}
+	} else {
+		errs = append(errs, errors.New("Unexpected log format"))
+	}
+
+	if len(errs) > 0 {
+		return componenterror.CombineErrors(errs)
+	}
+
 	return nil
 }
 
-func (se *sumologicexporter) ConsumeMetrics(_ context.Context, _ pdata.Metrics) error {
+// Send sends data to sumologic
+func (se *sumologicexporter) send(pipeline string, body string, fields string) error {
+	client := &http.Client{
+		Timeout: se.config.TimeoutSettings.Timeout,
+	}
+
+	// Add headers
+	req, _ := http.NewRequest("POST", se.config.URL, strings.NewReader(body))
```

Don't ignore the errors

---

done